File: apps/party-time-backend/src/main/java/com/partytime/jpa/mapper/AccountMapper.java
package com.partytime.jpa.mapper;

import com.partytime.api.dto.account.AccountDTO;
import com.partytime.jpa.entity.Account;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class AccountMapper {

    public static AccountDTO map(Account account) {
        return AccountDTO.builder()
            .id(account.getId())
            .name(account.getName())
            .email(account.getEmail())
            .emailVerified(account.isEmailVerified())
            .build();
    }

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/mapper/AccountMapper.kt
package com.partytime.jpa.mapper

import com.partytime.api.dto.account.AccountDTO
import com.partytime.jpa.entity.Account

/**
 * Transforms an [Account] entity into a (serializable) [AccountDTO]
 * All outbound account information has been saved to database and therefore has a not-null id.
 *
 * @receiver transformation subject
 * @return transformation result, ready for serialization and transmission to the client
 */
fun Account.toAccountDTO(): AccountDTO = AccountDTO(id!!, name, email)
